main.tf

```diff
@@ -7,9 +7,18 @@ locals {
     playbook_dir  = var.playbook_dir
     playbook_file = var.playbook_file
     playbook_repo = var.playbook_repo
+    repo_host     = try(local.repo_parts.host, null)
+    repo_port     = try(local.repo_parts.port, 22)
     ssh_key_name  = try(data.aws_secretsmanager_secret.ssh_key[0].name, null)
   })
 
+  repo_parts = try(
+    regex(
+      "^(?P<protocol>\\w+)://(?P<user>\\w+)@(?P<host>[\\w\\._-]+)(?::(?P<port>\\d+))?/(?P<user>[\\w_-]+)/(?P<repo>[\\w_-]+).git$",
```

Comment:
Given the design of the module, it appears this won't work on `https`. If that's the case, it might be nice to hardcode the protocol to ssh so that it breaks in a more straightforward way if someone passes `https`.

Reply:
https will work fine if it's a public repo (as many of ours are)